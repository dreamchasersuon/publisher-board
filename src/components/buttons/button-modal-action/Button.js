import React from 'react';
import PropTypes from 'prop-types';
import './Buttons.css';

export default function Button({ onPress, label, testId }) {
  return (
    <button
      data-testid={testId}
      onClick={onPress}
      type="button"
      className="Modal-action"
    >
      {label}
    </button>
  );
}

Button.propTypes = {
  onPress: PropTypes.func,
  label: PropTypes.string,
  testId: PropTypes.string,
};
