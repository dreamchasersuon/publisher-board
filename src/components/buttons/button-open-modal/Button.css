.Header-button {
  border: 1px solid #e5e5e5;
  border-radius: 4px;
  color: #666;
  background-color: transparent;
  margin-left: auto;
  outline: none;
  cursor: pointer;
  transition: all 0.6s cubic-bezier(0.39, 0.575, 0.565, 1);
}

.Header-button:hover {
  border-color: black;
  color: black;
}

.Popover-button {
  padding: 10px 20px;
  font-size: 14px;
  color: #444;
  text-align: left;
  border: none;
  outline: none;
  border-bottom: 1px solid #e5e5e5;
  cursor: pointer;
  background-color: white;
  transition: all 0.6s cubic-bezier(0.39, 0.575, 0.565, 1);
}

.Popover-button:last-child {
  border: none;
}

.Popover-button:hover {
  background-color: #fafafa;
}
