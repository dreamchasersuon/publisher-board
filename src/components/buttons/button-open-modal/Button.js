import React from 'react';
import PropTypes from 'prop-types';
import './Button.css';

export default function Button({
  title,
  onPress,
  dataName,
  className,
  testId,
}) {
  return (
    <button
      data-testid={testId}
      data-name={dataName}
      onMouseDown={onPress}
      className={className}
    >
      {title}
    </button>
  );
}

Button.propTypes = {
  title: PropTypes.string,
  onPress: PropTypes.func,
  dataName: PropTypes.string,
  className: PropTypes.string,
  testId: PropTypes.string,
};
