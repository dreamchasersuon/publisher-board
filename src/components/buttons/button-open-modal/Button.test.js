import React from 'react';
import snapshotDiff from 'snapshot-diff';
import Button from './Button';

const handleOnPress = jest.fn();

test('renders modal buttons with different states', () => {
  expect(
    snapshotDiff(
      <Button
        onPress={handleOnPress}
        title="Add"
        className="Header-button"
        dataName="addUser"
      />,
      <Button
        onPress={handleOnPress}
        title="Update User"
        className="Popover-button"
        dataName="updateUser"
      />,
    ),
  ).toMatchSnapshot();
});
