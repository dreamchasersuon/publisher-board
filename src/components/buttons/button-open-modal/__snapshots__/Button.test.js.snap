// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders modal buttons with different states 1`] = `
"Snapshot Diff:
- <Button className=\\"Header-button\\" dataName=\\"addUser\\" onPress={[Function mockConstructor]} title=\\"Add\\" />
+ <Button className=\\"Popover-button\\" dataName=\\"updateUser\\" onPress={[Function mockConstructor]} title=\\"Update User\\" />

  <button
-   className=\\"Header-button\\"
-   data-name=\\"addUser\\"
+   className=\\"Popover-button\\"
+   data-name=\\"updateUser\\"
    onMouseDown={[Function mockConstructor]}
  >
-   Add
+   Update User
  </button>"
`;
